app/models/sale.rb

```diff
@@ -0,0 +1,28 @@
+class Sale < ActiveRecord::Base
+  belongs_to :vendor
+  belongs_to :product
+  validate :monetary_value
```

Comment:
You can validate the numericality of the value which will simplify your validation method.

http://edgeguides.rubyonrails.org/active_record_validations.html#numericality